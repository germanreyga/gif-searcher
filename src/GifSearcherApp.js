import React, { useState } from "react";
import { CategoryAdder } from "./components/CategoryAdder";
import { GifsGrid } from "./components/GifsGrid";

export function GifSearcherApp({ defaultCategories = [] }) {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifSearcherApp</h2>
      <hr />
      <CategoryAdder setCategories={setCategories} />
      {categories.map((category) => {
        return <GifsGrid key={category} category={category} />;
      })}
    </>
  );
}
