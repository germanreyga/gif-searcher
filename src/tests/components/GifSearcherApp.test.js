import "@testing-library/jest-dom";
import React from "react";
import { shallow } from "enzyme";
import { GifSearcherApp } from "../../GifSearcherApp";

describe("testing the <GifSeacherApp /> component", () => {
  test("should show the component correctly", () => {
    const wrapper = shallow(<GifSearcherApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a list of categories", () => {
    const categories = ["Naruto", "Hunter x Hunter"];
    const wrapper = shallow(<GifSearcherApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifsGrid").length).toBe(categories.length);
  });
});
