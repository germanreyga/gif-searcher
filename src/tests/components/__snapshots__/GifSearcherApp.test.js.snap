// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the <GifSeacherApp /> component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifSearcherApp
  </h2>
  <hr />
  <CategoryAdder
    setCategories={[Function]}
  />
  <GifsGrid
    category="Naruto"
    key="Naruto"
  />
  <GifsGrid
    category="Hunter x Hunter"
    key="Hunter x Hunter"
  />
</Fragment>
`;

exports[`testing the <GifSeacherApp /> component should show the component correctly 1`] = `
<Fragment>
  <h2>
    GifSearcherApp
  </h2>
  <hr />
  <CategoryAdder
    setCategories={[Function]}
  />
</Fragment>
`;
